### PR TITLE
233 refactor browser

### PR DESCRIPTION
**Pull request by chStaiger**
I am not completely finished, the up and download buttons and their functionality need to be revised. But input on the rest is already very warmely welcome!!!!

- More responsiveness during browsing, caching implemented through flags
- Remove button to top
- ACL edits dependiing now on options specific to collection or data object
- Port sync to ibridges 1.0.1

### CODE REVIEW

**Review by qubixes — CHANGES_REQUESTED** [review]
The browser is so snappy since the last time I showed it off! I don't know if it's in this PR, but I love it!

File: ibridgesgui/browser.py [hunk 1]
```diff
@@ -39,98 +38,108 @@ def __init__(self, session, app_name):
...
+        # reset the params to load info tabs
+        self.last_selected_row = -1
+        self.current_selected_row = -1
+        self.updated_info_tabs = []
         self.load_browser_table()
 
-    def set_parent(self):
+    def set_input_path_to_home(self):
+        """Reset browser table to home."""
+        self.update_input_path(self.home_coll)
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/browser.py:103]: The little home button is also using thos function ... so yes, we need it in two places.

File: ibridgesgui/browser.py [hunk 2]
```diff
@@ -39,98 +38,108 @@ def __init__(self, session, app_name):
...
+
+    def set_input_path_to_parent(self):
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/browser.py:105]: But this one is also used by a button, the little up-arrow button.

File: ibridgesgui/browser.py [hunk 3]
```diff
@@ -150,33 +159,24 @@ def file_upload(self):
...
                     if button_reply == PyQt6.QtWidgets.QMessageBox.StandardButton.Yes:
-                        if Path(download_dir).joinpath(item_name).exists() and not overwrite:
-                            raise FileExistsError
                         self.setCursor(PyQt6.QtGui.QCursor(PyQt6.QtCore.Qt.CursorShape.BusyCursor))
                         self.logger.info(
-                            "Downloading %s to %s, overwrite %s", path, download_dir, str(overwrite)
+                            "Downloading %s to %s", path, download_dir
                         )
-                        download(self.session, path, download_dir, overwrite=overwrite)
+                        download(self.session, path, download_dir, overwrite=True)
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/browser.py]: For convenience during the development ;) It is outdated wuth the latest commits.

File: ibridgesgui/threads.py [hunk 4]
```diff
@@ -50,15 +51,15 @@ class TransferDataThread(QThread):
...
         """Pass parameters.
 
         ienv_path : Path
             path to the irods_environment.json to create a new session.
         logger : logging.Logger
             Logger
-        diffs : dict
-            A dict object containing four keys:
+        ops : Opertions
+            An object containing four attributes:
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/threads.py]: I adjusted the doc string and referred to the ibridges docs.

**Comment by chStaiger** [comment]
I finished the makeover of the upload and download buttons. I am not completely happy yet:
- The QDialog window stays on top of the main window and cannot be sent to the backend
- As long as the QDialog is open one cannot use the other features of the browser

As a workaround I made a utton to close the window while the data transfers are running in the backend. The logs accumulate the information what was successful and what not.

I think in a new PR we can have a look how to create a process ID for those closed windows and accumulate the concerning logging in a report.

**Review by qubixes — COMMENTED** [review]
I have found two things: When downloading a folder, the transfers are all done separately according to the log. This might be a bit slow? The metadata checkbox in the up/download is also unclear, it just gives the file that it will be saved to I suppose?

File: ibridgesgui/gui_utils.py [hunk 5]
```diff
@@ -82,25 +83,15 @@ def prep_session_for_copy(session, error_label) -> pathlib.Path:
...
-    elif isinstance(destination, IrodsPath):
-        print("not implemented yet")
-    return combined_diffs
+def combine_operations(operations: list[Operations]) -> Operations:
+    """Combine the operations of several upload or download dry-runs."""
+    ops = operations[0]
+    ops.create_dir = set().union(*[o.create_dir for o in operations])
+    ops.create_collection = set().union(*[o.create_collection for o in operations])
+    ops.download = set().union(*[o.download for o in operations])
+    ops.upload = set().union(*[o.upload for o in operations])
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/gui_utils.py]: Done!!! Thanks for pointing it out.

**Pull request merged**